cmd/genji/shell/shell.go

```diff
@@ -123,6 +125,16 @@ func Run(opts *Options) error {
 	return sh.dumpHistory()
 }
 
+func (sh *Shell) registerCommandSuggestion() {
```

Comment:
I don't think we need to register this and keep the state. We should simply rename it `getCommandSuggestions` and return the suggestions based on the `commands` map.
The `expected` slice below is duplicating the commands information which is not ideal, let's just use the global one.